encoder/vaapiencoder_base.cpp

```diff
@@ -88,7 +88,9 @@ YamiStatus VaapiEncoderBase::start(void)
 
 void VaapiEncoderBase::flush(void)
 {
-    AutoLock l(m_lock);
+    /* Current version of VaapiEncoderBase::flush is empty fucntion
+     * But we may add something in future.All derive class need call this in derive::flush()
+     */
```

Comment:
One way to enforce the `base::flush()` to be called is to "not allow" the derived class to override it.  Add an empty "protected" `virtual doFlush() { }` to the base class and derived classes can implement that if they want, instead.  Then, in the "public" `base::flush()` method we just call `doFlush()`.  This way, the derived class does not have to remember to call `base::flush()`.